modules/home-manager/waybar.nix

```diff
@@ -0,0 +1,40 @@
+{ config
+, lib
+, sources
+, ...
+}:
+let
+  cfg = config.programs.waybar.catppuccin;
+  enable = cfg.enable && config.programs.waybar.enable;
+in
+{
+  options.programs.waybar.catppuccin = (lib.ctp.mkCatppuccinOpt "waybar") // {
+    styleFile = lib.mkOption {
+      type = lib.types.path;
+      default = "${sources.waybar}/themes/${cfg.flavour}.css";
+      description = "Path to CSS file containing color variable definitions";
+    };
+
+    prependImport = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = "Whether to prepend `@import \"$${styleFile}\";` when `programs.waybar.style` is set to a string.";
+    };
+
+    createLink = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = "Whether to create a symlink `~/.config/waybar/catppuccin.css` pointing to `styleFile`.";
+    };
```

Comment:
i think one of these should be `true` by default, as all other modules will apply something when enabled individually or with the new global option; this is essentially no-op with `catppuccin.enable = true`. i also don't see a reason why both of these should ever be enabled together, so an assertion would be good (unless i'm not thinking of some specific case. i don't use waybar personally lol)

Reply:
Replaced these by an enum.

Reply:
Also added a warning if `prependImport` has no effect. (Can replace with an assertion if you prefer that.)